File: src/leetcode/python/medium/UniquePaths_62.py
class UniquePaths_62(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        maze = [[1 for x in range(n)] for x in range(m)]
        for i in range(1, m):
            for j in range(1, n):
                maze[i][j] = maze[i-1][j] + maze[i][j-1]
        return maze[-1][-1]

File: src/leetcode/java/UniquePaths_62.java
package leetcode.java;

public class UniquePaths_62 {
  public int uniquePaths(int m, int n) {
    if (m == 1 || n == 1) return 1;

    int[][] paths = new int[m][n];
    for (int i = 0; i < n; i++) paths[0][i] = 1;
    for (int j = 0; j < m; j++) paths[j][0] = 1;
    for (int i = 1; i < m; i++) {
      for (int j = 1; j < n; j++) {
        paths[i][j] = paths[i - 1][j] + paths[i][j - 1];
      }
    }
    return paths[m - 1][n - 1];
  }
}
